module/zfs/spa_config.c

```diff
@@ -167,14 +167,9 @@ spa_config_write(spa_config_dirent_t *dp, nvlist_t *nvl)
 	/*
 	 * Pack the configuration into a buffer.
 	 */
-	VERIFY(nvlist_size(nvl, &buflen, NV_ENCODE_XDR) == 0);
-
-	buf = vmem_alloc(buflen, KM_SLEEP);
+	buf = fnvlist_pack(nvl, &buflen);
 	temp = kmem_zalloc(MAXPATHLEN, KM_SLEEP);
```

Comment:
This minor restructuring is going to result in us using kmem_alloc() instead of vmem_alloc().  This poses a little bit of a problem since it's possible this allocation can be fairly large (40744 bytes as shown in #3251) which is why it was moved to vmem_alloc().  The SPL will generate console warnings for any allocations which exceed the default 32k threshold.  We need to preserve the vmem_alloc() / vmem_free() here.